packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -2968,6 +3017,8 @@ export default abstract class SqlIntegration
             `WHERE ((_TABLE_SUFFIX BETWEEN '{{date startDateISO "yyyyMMdd"}}' AND '{{date endDateISO "yyyyMMdd"}}') OR
  (_TABLE_SUFFIX BETWEEN 'intraday_{{date startDateISO "yyyyMMdd"}}' AND 'intraday_{{date endDateISO "yyyyMMdd"}}')) 
 AND event_name = '${eventName}'`,
+          getFactTableWhereClause: (eventName: string) =>
+            `WHERE event_name = '${eventName}'`,
```

Comment:
I don't really understand what's going on here.  There's ` getDateLimitClause`, but then the `getMetricWhereClause` duplicates all of that logic, but the `getFactTableWhereClause` does not

Reply:
So... I think there are two things happening here.

1. I think the `getMetricWhereClause` for `GA4` that contains the date filtering is a bug. We only want the 7 day date limit when getting the tracked events for a data source, not when actually running the metric - we instead when running the metric query we want to use the `metricAnalysisDays`.

I think this isn't causing an issue because when we go to run the metric query, the date fields used in the `getMetricWhereClause` for GA4 are essentially duplicating the `metricAnalysisDays` logic. I took a screenshot below of the different in the queries when the `getMetricWhereClause` contains the date filtering, and when it doesn't.

<img width="974" alt="Screen Shot 2024-06-25 at 8 36 38 AM" src="https://github.com/growthbook/growthbook/assets/75274610/eb2e5336-3c14-4dd2-8715-2feee8e53b22">

Given that, we should be able to remove the date filtering from GA4's `getMetricWhereClause` - and when doing that, that means for all of the supported schema types, the values of `getMetricWhereClause` and the `getFactTableWhereClause` should be the same, so we can combine them into a single property. Maybe `getEventFilterWhereClause`?